### PR TITLE
[Incubator][VC] Using the signature of the tenant apiserver certificate to identify the request sender

### PR DESCRIPTION
**Pull request by charleszheng44**
Using the signature of the tenant apiserver certificate to identify the sender of a request received by vn-agent

1. add the annotation of the certificate's signature to VC CR after it
is created.
2. when receiving a request, vn-agent extracts the signature from the
client certificate and find the VC that has the same signature.
4. implement the `GetClusterCertificate` function for
`master_provisioner_native`
3. add RBAC rights to vn-agent-role for accessing virtualcluster CR

### CODE REVIEW

File: incubator/virtualcluster/pkg/controller/constants/constants.go [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package constants
```
Comment (zhuangqh) [inline incubator/virtualcluster/pkg/controller/constants/constants.go:1]: please add the file header

**Comment by zhuangqh** [comment]
lgtm overall

the kube-apiserver certificate is the same as kubelet-client certificate in minikube cluster, right?

**Comment by Fei-Guo** [comment]
> lgtm overall
> 
> the kube-apiserver certificate is the same as kubelet-client certificate in minikube cluster, right?

It seems not to be the case. We need to find some other methods.

**Comment by charleszheng44** [comment]
> lgtm overall
> 
> the kube-apiserver certificate is the same as kubelet-client certificate in minikube cluster, right?

Yes. But if we use cloud mode, the `kube-apiserver` certificate and the `kubelet-client-certificate` are different.

**Pull request closed without merging**